core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -270,7 +270,11 @@ object DumpLogSegments {
 
             var prefix = s"${RecordIndent} "
             if (!skipRecordMetadata) {
-              print(s"${prefix}offset: ${record.offset} isValid: ${record.isValid} crc: ${record.checksumOrNull}" +
+              val checksum = record match {
+                case r: AbstractLegacyRecordBatch => r.checksum().toString
+                case _ => "null"
+              }
```

Comment:
I think we should skip the checksum string altogether at record level for message format v2 and later.

Reply:
sure